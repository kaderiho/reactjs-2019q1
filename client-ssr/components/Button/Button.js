"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var Button = function Button(_ref) {
  var _classNames;

  var _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? 'contained' : _ref$variant,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'primary' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'medium' : _ref$size,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'button' : _ref$type,
      className = _ref.className,
      children = _ref.children,
      onClick = _ref.onClick,
      name = _ref.name;
  var buttonClass = (0, _classnames.default)('btn', className, "btn--".concat(size), (_classNames = {}, (0, _defineProperty2.default)(_classNames, "btn--".concat(variant), variant), (0, _defineProperty2.default)(_classNames, "btn--".concat(color), color), _classNames));
  return _react.default.createElement("button", {
    className: buttonClass,
    onClick: onClick,
    type: type,
    name: name
  }, children);
};

var _default = Button;
exports.default = _default;