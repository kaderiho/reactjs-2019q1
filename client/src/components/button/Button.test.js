import React from 'react';
import Button from './Button';
import { shallow, mount } from 'enzyme';

const buttonClickFn = jest.fn();

describe('Button component', () => {
    it('Should render correctly with default props', () => {
        const wrapper = shallow(<Button>My button</Button>);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correctly with passed props', () => {
        const buttonProps = {
            variant: 'transparent',
            children: 'My button',
            color: 'secondary',
            size: 'small'
        };
        const wrapper = shallow(<Button {...buttonProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('onClick handler should have been called', () => {
        const wrapper = mount(<Button onClick={buttonClickFn} />);

        wrapper.find('button.btn').simulate('click');

        expect(buttonClickFn).toHaveBeenCalled();
    });
});


