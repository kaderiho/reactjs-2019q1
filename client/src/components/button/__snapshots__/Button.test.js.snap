// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render correctly with default props 1`] = `
<button
  className="btn btn--medium btn--contained btn--primary"
  type="button"
>
  My button
</button>
`;

exports[`Button component Should render correctly with passed props 1`] = `
<button
  className="btn btn--small btn--transparent btn--secondary"
  type="button"
>
  My button
</button>
`;
